src/sentry/users/models/lostpasswordhash.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+from datetime import timedelta
+from typing import TYPE_CHECKING, Any
+
+from django.conf import settings
+from django.db import models
+from django.urls import reverse
+from django.utils import timezone
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import FlexibleForeignKey, Model, control_silo_model, sane_repr
+from sentry.users.services.user.model import RpcUser
+from sentry.utils.http import absolute_uri
+from sentry.utils.security import get_secure_token
+
+if TYPE_CHECKING:
+    from sentry.models.user import User
+    from sentry.users.services.lost_password_hash import RpcLostPasswordHash
+
+
+@control_silo_model
+class LostPasswordHash(Model):
+    __relocation_scope__ = RelocationScope.Excluded
+
+    user = FlexibleForeignKey(settings.AUTH_USER_MODEL, unique=True)
+    hash = models.CharField(max_length=32)
+    date_added = models.DateTimeField(default=timezone.now)
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_lostpasswordhash"
+
+    __repr__ = sane_repr("user_id", "hash")
+
+    def save(self, *args: Any, **kwargs: Any) -> None:
+        if not self.hash:
+            self.set_hash()
+        super().save(*args, **kwargs)
+
+    def set_hash(self) -> None:
+        self.hash = get_secure_token()
+
+    def is_valid(self) -> bool:
+        return self.date_added > timezone.now() - timedelta(hours=1)
+
+    @classmethod
+    def send_recover_password_email(cls, user: User, hash: str, ip_address: str) -> None:
+        extra = {
+            "ip_address": ip_address,
+        }
+        cls._send_email("recover_password", user, hash, extra)
+
+    @classmethod
+    def send_relocate_account_email(
+        cls, user: User | RpcUser, hash: str, orgs: Iterable[str]
+    ) -> None:
+        cls._send_email("relocate_account", user, hash, {"orgs": orgs})
+
+    @classmethod
+    def _send_email(cls, mode: str, user: User | RpcUser, hash: str, extra: dict[str, Any]) -> None:
```

Comment:
the extra is a bit odd since it's basically **kwargs

Reply:
Yeah, we don't need to change it though. We have a similar pattern in a few other email builder classes that take a mapping of data to pass into templates that are being rendered.